### DISCUSSION TITLE
Application calls home several times to 172.16.45.166

### DISCUSSION THREAD

**Discussion by pbarney** [timschneeb/GalaxyBudsClient#313] [Q&A] [answered]
Just curious why the app is making outbound internet access requests.

It appears to be a Cloudflare IP address.

**Comment by timschneeb** [comment, marked as answer]
These features of this app access the internet:

* **Application update notifications**
Self-explanatory; example: https://timschneeberger.me/updates/galaxybudsclient/appcast.xml
Requested on every launch and then approx. every 2 hours

* **Firmware updates**
The app refreshes its firmware index on each launch. I'm keeping a collection of new and old firmware binaries on my server to allow downgrading. (The source code for that API is available here: https://github.com/ThePBone/FirmwareProviderAPI)

* **Crowdsourcing serial protocol data**
This is optional and can be disabled on the settings page. To support all Galaxy Buds models (including those I don't own), the app can selectively collect fragments of the binary data stream exchanged over Bluetooth if necessary. This helps me figure out more easily how Samsung's proprietary configuration protocol was changed/updated if something broke inside my app that started causing crashes or unintended behavior when interacting with the Buds via Bluetooth.

* **Crash reports** (via <https://sentry.io>)
This is also optional and can be disabled on the settings page as well. 

The first three features connect to my server that is hosted behind a Cloudflare firewall. You probably found web requests for one of these things.
